Repositories/ItemRepository/ItemRepository.cs

```diff
@@ -0,0 +1,39 @@
+﻿using DomainModel;
+using System.Linq;
+
+namespace Repositories.ItemRepository
+{
+    public class ItemRepository : IItemRepository
+    {
+        DataContext context;
+
+        public ItemRepository(DataContext datacontext)
+        {
+            context = datacontext;
+        }
+
+        public IQueryable<Item> All()
+        { 
+            return context.Items.AsQueryable();
+        }
+
+        public void Delete(int id)
+        {
+            context.Items.RemoveAt(id - 1);
+            context.Items = context.Items;
+        }
+
+        public void Save(Item item)
+        {
+            if (item != null)
```

Comment:
item here will never be null because both ItemService Add and Update is passing a new instance of Item. Can you redesign this logic? A better approach would be fetching the item from the context items first using ItemID before this line.